src/lib/io/sol-iio.c

```diff
@@ -690,6 +711,19 @@ sol_iio_open(int device_id, const struct sol_iio_config *config)
         return NULL;
     }
 #endif
+    // find the open list 
+    SOL_LIST_FOREACH (&iio_open_list, it) {
+        entry = SOL_LIST_GET_CONTAINER(it, struct open_list, list);
+        SOL_INF("");
+        if (device_id == entry->device->device_id) {
+            if (0 == memcmp(config, entry->config, sizeof(struct sol_iio_config))) {// the same
+                return entry->device;
+            } else { // same device_id but different config
+            SOL_WRN("Could not open IIO device%d with different config", device_id);
```

Comment:
Please, make this message more informative. Say that device is already open, but couldn't reuse it because of different config.